### PR TITLE
Add og images

### PR DESCRIPTION
**Pull request by mashehu**
needs SSR to read in searchParams, but we could also build them completely statically by going through the front matter.

only for the docs pages for now

### CODE REVIEW

File: src/pages/docs/[...slug].astro [hunk 1]
```diff
@@ -69,6 +70,9 @@ for (const match of matches) {
   subtitle={frontmatter.description}
   headings={headings}
   md_github_url={gh_url}
+  image={`/og.png?title=${encodeURIComponent(frontmatter.title)}&subtitle=${encodeURIComponent(
```
Comment (ewels) [inline src/pages/docs/[...slug].astro:73]: Would be good to be able to optionally provide this in the markdown frontmatter, then default to what you have here 👍🏻

Comment (ewels) [reply src/pages/docs/[...slug].astro:73]: eg. something like:
```suggestion
  image={frontmatter.social_image ? frontmatter.social_image : `/og.png?title=${encodeURIComponent(frontmatter.title)}&subtitle=${encodeURIComponent(
```

**Comment by mashehu** [comment]
NB: change `site` in astro.config.mjs to `https://astro--multiqc.netlify.app` before merging

File: astro.config.mjs [hunk 2]
```diff
@@ -30,7 +30,7 @@ const AnchorLinkIcon = h(
 
 // https://astro.build/config
 export default defineConfig({
-  site: "https://multiqc.info",
+  site: "https://deploy-preview-38--multiqc.netlify.app", // TODO: switch back to "https://multiqc.info"
```
Comment (ewels) [inline astro.config.mjs:33]:
```suggestion
  site: "https://astro--multiqc.netlify.app", // TODO: switch back to "https://multiqc.info"
```

**Review by ewels — APPROVED** [review]

**Review by ewels — APPROVED** [review]
Awesome work 🎉

**Pull request merged**